omnigibson/utils/usd_utils.py

```diff
@@ -568,12 +748,291 @@ def get_world_pose_with_scale(cls, prim_path):
+        if "coriolis_and_centrifugal_forces" not in self._read_cache:
+            self._read_cache["coriolis_and_centrifugal_forces"] = self._view.get_coriolis_and_centrifugal_forces()
+
+        idx = self._idx[prim_path]
+        return self._read_cache["coriolis_and_centrifugal_forces"][idx]
+
+    def get_link_relative_position_orientation(self, prim_path, link_name):
+        if "link_transforms" not in self._read_cache:
+            self._read_cache["link_transforms"] = self._view.get_link_transforms()
+
+        idx = self._idx[prim_path]
+        link_idx = self._link_idx[idx][link_name]
+        pose = self._read_cache["link_transforms"][idx][link_idx]
+        pos, orn = pose[:3], pose[3:]
+
+        # Get the root world transform too
+        world_pos, world_orn = self.get_position_orientation(prim_path)
+
+        # Compute the relative position and orientation
+        return T.relative_pose_transform(pos, orn, world_pos, world_orn)
+
+    def get_link_relative_linear_velocity(self, prim_path, link_name):
+        if "link_velocities" not in self._read_cache:
+            self._read_cache["link_velocities"] = self._view.get_link_velocities()
+
+        idx = self._idx[prim_path]
+        link_idx = self._link_idx[idx][link_name]
+        vel = self._read_cache["link_velocities"][idx][link_idx]
+        linvel = vel[:3]
+
+        # Get the root world transform too
+        _, world_orn = self.get_position_orientation(prim_path)
+
+        # Compute the relative position and orientation
+        return T.quat2mat(world_orn).T @ linvel
+
+    def get_link_relative_angular_velocity(self, prim_path, link_name):
+        if "link_velocities" not in self._read_cache:
+            self._read_cache["link_velocities"] = self._view.get_link_velocities()
+
+        idx = self._idx[prim_path]
+        link_idx = self._link_idx[idx][link_name]
+        vel = self._read_cache["link_velocities"][idx][link_idx]
+        angvel = vel[3:]
+
+        # Get the root world transform too
+        _, world_orn = self.get_position_orientation(prim_path)
+
+        # Compute the relative position and orientation
+        return T.mat2euler(T.quat2mat(world_orn).T @ T.euler2mat(angvel))
+
+    def get_jacobian(self, prim_path):
+        if "jacobians" not in self._read_cache:
+            self._read_cache["jacobians"] = self._view.get_jacobians()
+
+        idx = self._idx[prim_path]
+        return self._read_cache["jacobians"][idx]
+
+    def get_relative_jacobian(self, prim_path):
+        jacobian = self.get_jacobian(prim_path)
+        ori_t = T.quat2mat(self.get_position_orientation(prim_path)[1]).T.astype(np.float32)
+        tf = np.zeros((1, 6, 6), dtype=np.float32)
+        tf[:, :3, :3] = ori_t
+        tf[:, 3:, 3:] = ori_t
+        return tf @ jacobian
+
+
+ControllableObjectViewAPI = BatchControlViewAPIImpl("/World/scene_*/controllable_*/base_link")
+DummyObjectViewAPI = BatchControlViewAPIImpl("/World/scene_*/dummy_*/base_link")
```

Comment:
maybe better naming scheme here -- DummyControllableObjectViewAPI? Still sounds bad tho...maybe let's discuss more in person

Reply:
@hang-yin this sgtm feel free to change